src/components/main-form.js

```diff
@@ -1,180 +1,192 @@
 // References:
 // https://www.bankrate.com/calculators/mortgages/new-house-calculator.aspx
 // https://www.nerdwallet.com/mortgages/mortgage-rates/30-year-fixed
+// https://michaelbluejay.com/house/15vs30.html
 
 import _ from 'lodash';
 import React from 'react';
 import numeral from 'numeral';
-import { InputGroup, InputGroupAddon, InputGroupText, Input } from 'reactstrap';
+import styled, { css } from "react-emotion";
+import { Input } from 'reactstrap';
 import { FV, PMT } from 'formulajs/lib/financial';
 
+
+const COMPOUND_FREQUENCY = 12;
+
+const o = {
+  interestAmt: 0,
+  interestRate: 0,
+  term: 0,
+  pmt: 0,
+  fv: 0
+};
+
+let o1 = Object.assign({}, o);
+let o2 = Object.assign({}, o);
+
+o1.interestRate = 4.3;
+o1.term = 30;
+
+o2.interestRate = 4;
+o2.term = 15;
+
+const Row = styled('div')`
+  display: flex;
+  flex-direction: row;
+`
+
 class MainForm extends React.Component {
 
   constructor() {
     super();
-
-    this.calculate = this.calculate.bind(this);
-    this.updateInput = this.updateInput.bind(this);
-    this.validateData = this.validateData.bind(this);
-
-    this.state = this.calculate({
-      investmentRate: 7,
-      mortgageAmt: 200000,
+    this.calculatePMT = this.calculatePMT.bind(this)
+    this.calculateInterestAmt = this.calculateInterestAmt.bind(this)
+    this.calculateFV = this.calculateFV.bind(this)
+    this.calculateOptCost = this.calculateOptCost.bind(this);
```

Comment:
```suggestion
    this.calculateOpportunityCost = this.calculateOpportunityCost.bind(this);
```

improves readability when we read this code again in a few weeks

Reply:
fine with me!